Add foo.txt and bar.txt to .gitignore

thomaswhiteway opened the pull request:
Some of the tests generate files called `foo.txt` and `bar.txt`, these should not get checked in.

## Motivation

I kept almost checking in `foo.txt` and `bar.txt` by mistake.

## Solution

Add them to `.gitignore`

carllerche commented:
Thanks for the Pr!

Instead, what if we found the tests that generated them and fixed those?

thomaswhiteway commented:
> Thanks for the Pr!
> 
> Instead, what if we found the tests that generated them and fixed those?

I'd thought there were a whole bunch of tests that used those files, but on closer inspection all but one of the references were in docs with `no_run` set.  I've fixed up that remaining case to create the files in a temporary directory that gets cleaned up when the test ends.  Is that better?

kleimkuhler commented on tokio/tests/fs_copy.rs:
```diff
@@ -1,15 +1,21 @@
 #![warn(rust_2018_idioms)]
 #![cfg(feature = "full")]
 
+use tempdir::TempDir;
```
`tempfile` is already a dependency so we should just use that instead.

```
let foo = NamedTempFile::new().unwrap();
..
foo.path()
```
